at-angular.ts

```diff
@@ -58,39 +63,77 @@ module at {
         (moduleName: string, serviceName: string): IClassAnnotationDecorator;
     }
 
-    export function service(moduleName: string, serviceName: string): at.IClassAnnotationDecorator {
-        return instantiate(moduleName, serviceName, 'service');
+    export function service(moduleName: string, serviceName?: string): at.IClassAnnotationDecorator {
+        return instantiate(moduleName, 'service', serviceName);
     }
 
     export interface IControllerAnnotation {
         (moduleName: string, ctrlName: string): IClassAnnotationDecorator;
     }
 
-    export function controller(moduleName: string, ctrlName: string): at.IClassAnnotationDecorator {
-        return instantiate(moduleName, ctrlName, 'controller');
+    export function controller(moduleName: string, ctrlName?: string): at.IClassAnnotationDecorator {
+        return instantiate(moduleName, 'controller', ctrlName);
     }
 
     export interface IDirectiveAnnotation {
         (moduleName: string, directiveName: string): IClassAnnotationDecorator;
     }
 
-    export function directive(moduleName: string, directiveName: string): at.IClassAnnotationDecorator {
+    export interface IDirectiveProperties extends angular.IDirective {
+        selector: string;
+    }
+
+    export function directive(
+        moduleName: string,
+        directiveSettings:  string|IDirectiveProperties
+    ): at.IClassAnnotationDecorator {
+
         return (target: any): void => {
-            let config: angular.IDirective;
-            const ctrlName: string = angular.isString(target.controller) ? target.controller.split(' ').shift() : null;
+
+            let config: IDirectiveProperties;
+            const ctrlName: string = angular.isString(target.controller)
+                ? target.controller.split(' ').shift()
+                : null;
+
+            if (typeof directiveSettings === 'string') {
+
+                // Retrocompatibilty
+
+                config = directiveProperties
+                    .reduce((config: angular.IDirective, property: string) => {
+                        return angular.isDefined(target[property]) ?
+                            angular.extend(config, {[property]: target[property]}) : config;
+                        }, {
+                            controller: target
+                    });
+
+                config.selector = directiveSettings;
+
+            } else {
+
+                // Generate config from Annotation configuration
+
+                config = <IDirectiveProperties> angular.copy(directiveSettings);
+
+                // Store FuncName as ControllerAs
+                config.controllerAs = getFuncName(target);
+                config.controller = target;
+
+                angular.forEach(directiveProperties, function(property: string): any {
+                    if (angular.isDefined(target[property])) {
+                        config[property] = target[property];
+                    }
+                });
+            }
```

Comment:
My pull request It's just proposal. You can get some lines or any. It's your choice and I understand that :)